### ISSUE TITLE
A task is missing. The pipeline references a task called 'AzureAppConfigurationPush' with version '7'. This indicates that the specific task version can not be found.

### ISSUE THREAD

**Issue by cvespa** [Azure/AppConfiguration#968]
Not sure if the Azure team is aware but we are running into the same issues as #959. It started happening after Sep 16, 2024 at 9:56 AM EDT

**Comment by ez5504** [comment]
@cvespa using v10 seems to work for us, despite the resolution for #959 stating that there were no plans to re-release v10...

**Comment by concreteutopia** [comment]
we are experiencing the same issue again with v7 of the push task and v10 appears to work for us.  we've got a much overdue deployment today and this will block our pipelines needs to be resolved in the next hour or so

**Comment by concreteutopia** [comment]
@MaryanneNjeri @jimmyca15 this issue has returned, please fix asap

**Comment by jimmyca15** [comment]
Under investigation.

**Comment by jimmyca15** [comment]
Please retry running your tasks at this point.

**Comment by cvespa** [comment]
> Please retry running your tasks at this point.

I was able to kick off my pipeline that uses this task

**Comment by cvespa** [comment]
After kick off the job failed for `##[error]wrong number of segments` We were seeing this error when #959 started but the Azure allowed the job to run

**Comment by jimmyca15** [comment]
@cvespa you're using v7 right?

**Comment by jimmyca15** [comment]
@ez5504 

Do you mean that new runs of an existing pipeline or working, or that creating a new pipeline entirely is working?

**Comment by cvespa** [comment]
> @cvespa you're using v7 right?

Yes.

> > Please retry running your tasks at this point.
> 
> We're seeing the same behavior as @cvespa. Running new pipelines with the v7 task seems to work, but anything that was out there before continues to fail with that `wrong number of segments` error.

let me try a whole new one

**Comment by concreteutopia** [comment]
kicking off a new pipeline with v7 works, but running a pending stage (we have approval gates) of a pipeline that earlier stages have run does not

**Comment by ez5504** [comment]
> @ez5504
> 
> Do you mean that new runs of an existing pipeline or working, or that creating a new pipeline entirely is working?

New runs of an existing pipeline. Old runs of an existing pipeline are failing i.e. redeploying a stage that already ran. We haven't tried creating new pipelines altogether but I wouldn't think we need to.

**Comment by jimmyca15** [comment]
> New runs of an existing pipeline

Got it.

> We haven't tried creating new pipelines altogether but I wouldn't think we need to.

No need, just wanted to clarify.

**Comment by cvespa** [comment]
> > @cvespa you're using v7 right?
> 
> Yes.
> 
> > > Please retry running your tasks at this point.
> > 
> > 
> > We're seeing the same behavior as @cvespa. Running new pipelines with the v7 task seems to work, but anything that was out there before continues to fail with that `wrong number of segments` error.
> 
> let me try a whole new one

Tring a `Run Pipeline` did not work for me still get the `##[error]wrong number of segments`

**Comment by ez5504** [comment]
We are now seeing issues with *both* v7 and v10...

**Comment by jimmyca15** [comment]
@ez5504

> We are now seeing issues with both v7 and v10...

There is no v10. There is only v7.

>Tring a Run Pipeline did not work for me still get the ##[error]wrong number of segments

Does the yml of your pipeline reference the task as

`AzureAppConfigurationPush`

or

`AzureAppConfiguration.azure-app-configuration-task-push.custom-build-release-task.AzureAppConfigurationPush`

**Comment by cvespa** [comment]
@jimmyca15  `AzureAppConfigurationPush`

**Comment by ez5504** [comment]
> @ez5504
> 
> > We are now seeing issues with both v7 and v10...
> 
> There is no v10. There is only v7.
> 
> @cvespa
> 
> > Tring a Run Pipeline did not work for me still get the ##[error]wrong number of segments
> 
> Does the yml of your pipeline reference the task as
> 
> `AzureAppConfigurationPush`
> 
> or
> 
> `AzureAppConfiguration.azure-app-configuration-task-push.custom-build-release-task.AzureAppConfigurationPush`

Maybe right now there is only v7, but v10 was working earlier and then seems to be broken again. 

As for our yml - we reference `AzureAppConfigurationPush@7` (or @10, for when we tried it and it was working)

**Comment by jimmyca15** [comment]
@cvespa 

Am I following right that you are blocked in any case and aren't able to get success at this point?

**Comment by cvespa** [comment]
> @cvespa
> 
> Am I following right that you are blocked in any case and aren't able to get success at this point?

Correct

**Comment by ez5504** [comment]
We've just in the last minute or so had success using v7 on both new and old runs of existing pipelines.

**Comment by jimmyca15** [comment]
I am suspecting this is some caching issue. We are working to verify.

**Comment by jimmyca15** [comment]
@cvespa 

Are you able to provide the URL of an impacted pipeline run for investigation?

**Comment by jimmyca15** [comment]
@concreteutopia 

Are you still experiencing issues?

**Comment by concreteutopia** [comment]
v7 appears to be working again in new/old pipelines

**Comment by jimmyca15** [comment]
Thank you for confirming.

**Comment by cvespa** [comment]
> @cvespa
> 
> Are you able to provide the URL of an impacted pipeline run for investigation?

Sorry, did not see this till now. It looks like it was a cache issues as I just ran a pipeline and it completed.

**Comment by ez5504** [comment]
@jimmyca15 So is the root cause for all of this some sort of caching issue? Or is there something bigger going on? And should we expect for this to happen again? It causes a great deal of disruption.

**Comment by jimmyca15** [comment]
@ez5504 

No, the root cause is not a caching issue. Details on why I mentioned caching below.

**Main issue**
The main issue here was receiving the message "`A task is missing. The pipeline references a task called 'AzureAppConfigurationPush' with version '7'. This indicates that the specific task version can not be found.`" when running a pipeline.

The root cause of this was a bad ADO deployment. The ADO team rolled back their release and that fixed this issue. 

**Secondary issue**

When the main issue was resolved some users started seeing an error that said "`##[error]wrong number of segments`". This was the issue that I described as a caching problem, because it took a bit for the resolution of the main issue to take affect universally. Until that happened this error could be seen.

---

> stating that there were no plans to re-release v10

Regarding this, this is still true. The App Configuration team has no plans to release v10 of the `AppConfiguration` or `AppConfigurationPush` tasks. This issue was not the result of App Configuration team rolling out v10 of these tasks, which was the root cause of a recent issue with the same error message linked in the original issue post (#959).

However, the App Configuration team will be releasing two new tasks named `AppConfigurationExport` and `AppConfigurationImport` as built-in tasks. They will be the successors of the `AppConfiguration` and `AppConfigurationPush` tasks respectively.
